executor/mem_reader.go

```diff
@@ -150,18 +174,15 @@ func (m *memIndexReader) getMemRows(ctx context.Context) ([][]types.Datum, error
 			}
 			return ret
 		})
-		return m.addedRows, err
```

Comment:
Should this be kept as `return nil, err`? In the case of the `m.compare` function gives an error in `slices.SortFunc()`?